theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -333,13 +342,17 @@ def _optimize_loop(
     # `track_best_solution` keeps a **detached** copy (as in no gradient info)
     # of the best variables found, but it is optional to avoid unnecessary copying
     # if this is not needed
+    #
+    # If `end_step_callback` is passed, it's called at the very end of each optimizer
+    # step, with four input arguments: (optimizer, info, delta, step_idx).
```

Comment:
each optimizer step is each iteration or each forward call i.e. at the termination condition (max_iters, etc)?

Reply:
yeah end_iter is clearer that it's at the end of iteration

Reply:
Renamed.